src/components/Hoverable/index.js

```diff
@@ -1,189 +1,211 @@
+ * parent. https://github.com/necolas/react-native-web/issues/1875
+ */
 
-    /**
-     * Sets the hover state of this component to true and execute the onHoverIn callback.
-     *
-     * @param {Boolean} isHovered - Whether or not this component is hovered.
-     */
-    setIsHovered(isHovered) {
-        if (this.props.disabled) {
-            return;
-        }
+const Hoverable = React.forwardRef(({disabled, onHoverIn, onHoverOut, children, shouldHandleScroll}, outerRef) => {
+    const [isHovered, setIsHovered] = useState(false);
 
-        /**
-         * Capture whther or not the user is hovering over the component.
-         * We will use this to determine if we should update the hover state when the user has stopped scrolling.
-         */
-        this.isHoveredRef = isHovered;
+    const isScrolling = useRef(false);
+    const isHoveredRef = useRef(false);
+    const ref = useRef(null);
 
-        /**
-         * If the isScrollingRef is true, then the user is scrolling and we should not update the hover state.
-         */
-        if (this.isScrollingRef && this.props.shouldHandleScroll && !this.state.isHovered) {
+    const updateIsHoveredOnScrolling = useCallback(
+        (hovered) => {
+            if (disabled) {
+                return;
+            }
+
+            isHoveredRef.current = hovered;
+
+            if (shouldHandleScroll && isScrolling.current) {
+                return;
+            }
+            setIsHovered(hovered);
+        },
+        [disabled, shouldHandleScroll],
+    );
+
+    useEffect(() => {
+        const unsetHoveredWhenDocumentIsHidden = () => document.visibilityState === 'hidden' && setIsHovered(false);
+
+        document.addEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
+
+        return () => document.removeEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
+    }, []);
+
+    useEffect(() => {
+        if (!shouldHandleScroll) {
             return;
         }
 
-        if (isHovered !== this.state.isHovered) {
-            this.setState({isHovered}, isHovered ? this.props.onHoverIn : this.props.onHoverOut);
-        }
-    }
+        const scrollingListener = DeviceEventEmitter.addListener(CONST.EVENTS.SCROLLING, (scrolling) => {
+            isScrolling.current = scrolling;
+            if (!scrolling) {
+                setIsHovered(isHoveredRef.current);
+            }
+        });
+
+        return () => scrollingListener.remove();
+    }, [shouldHandleScroll]);
 
     /**
      * Checks the hover state of a component and updates it based on the event target.
      * This is necessary to handle cases where the hover state might get stuck due to an unreliable mouseleave trigger,
      * such as when an element is removed before the mouseleave event is triggered.
      * @param {Event} e - The hover event object.
      */
-    checkHover(e) {
-        if (!this.wrapperView || !this.state.isHovered) {
+    const unsetHoveredIfOutside = useCallback(
```

Comment:
This function could be inside useEffect with mouseover listener as it is not used anywhere else

Reply:
Done ✅